test/e2e/gardener/shoot/internal/node/node.go

```diff
@@ -157,52 +160,64 @@ func createOrUpdateNodeCriticalManagedResource(ctx context.Context, seedClient,
 	}
 
 	data, err := managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer).AddAllAndSerialize(daemonSet)
-	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+	Expect(err).NotTo(HaveOccurred())
 
 	secretName, secret := managedresources.NewSecret(seedClient, namespace, name, data, true)
 	managedResource := managedresources.NewForShoot(seedClient, namespace, name, managedresources.LabelValueGardener, false).WithSecretRef(secretName)
 
-	ExpectWithOffset(1, secret.AddLabels(getLabels(name)).Reconcile(ctx)).To(Succeed())
-	ExpectWithOffset(1, managedResource.WithLabels(getLabels(name)).Reconcile(ctx)).To(Succeed())
+	Expect(secret.AddLabels(getLabels(name)).Reconcile(ctx)).To(Succeed())
+	Expect(managedResource.WithLabels(getLabels(name)).Reconcile(ctx)).To(Succeed())
 
 	By("Wait for DaemonSet to be applied in shoot")
-	EventuallyWithOffset(1, func(g Gomega) string {
+	Eventually(func(g Gomega) string {
 		g.Expect(shootClient.Get(ctx, client.ObjectKeyFromObject(daemonSet), daemonSet)).To(Succeed())
 		return daemonSet.Spec.Template.Spec.Containers[0].Image
-	}).WithContext(ctx).WithTimeout(5 * time.Minute).Should(Equal(image))
+	}).Should(Equal(image))
 
 	return daemonSet
 }
 
 func waitForDaemonSetToBecomeHealthy(ctx context.Context, shootClient client.Client, daemonSet *appsv1.DaemonSet) {
+	GinkgoHelper()
+
 	By("Wait for DaemonSet to become healthy")
-	EventuallyWithOffset(1, func(g Gomega) {
+	Eventually(func(g Gomega) {
 		g.Expect(shootClient.Get(ctx, client.ObjectKeyFromObject(daemonSet), daemonSet)).To(Succeed())
 		g.Expect(health.CheckDaemonSet(daemonSet)).To(Succeed())
-	}).WithContext(ctx).WithTimeout(5 * time.Minute).Should(Succeed())
+	}).Should(Succeed())
 }
 
 func patchCSINodeObjectWithRequiredDriver(ctx context.Context, shootClient client.Client) {
+	GinkgoHelper()
+
 	csiNodeList := &storagev1.CSINodeList{}
 
-	EventuallyWithOffset(1, func(g Gomega) {
+	Eventually(func(g Gomega) {
```

Comment:
Can you split up this function into "wait for CSINode object" and "patch CSINode object"?